crates/bevy_core/src/time/stopwatch.rs

```diff
@@ -169,3 +169,160 @@ impl Stopwatch {
+/// assert_eq!(stopwatch.elapsed(), 1);
+///
+/// stopwatch.reset(); // reset the stopwatch
+/// assert!(stopwatch.paused());
+/// assert_eq!(stopwatch.elapsed(), 0);
+/// ```
+#[derive(Clone, Debug, Default, Reflect)]
+#[reflect(Component)]
+pub struct DiscreteStopwatch {
+    elapsed: u64,
+    paused: bool,
+}
+
+impl DiscreteStopwatch {
+    /// Create a new unpaused `Stopwatch` with no elapsed increments.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let stopwatch = DiscreteStopwatch::new();
+    /// assert_eq!(stopwatch.elapsed(), 0);
+    /// assert_eq!(stopwatch.paused(), false);
+    /// ```
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Returns the elapsed increments since the last [`reset`](DiscreteStopwatch::reset)
+    /// of the stopwatch.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// use std::time::Duration;
+    /// let mut stopwatch = DiscreteStopwatch::new();
+    /// stopwatch.tick(1);
+    /// assert_eq!(stopwatch.elapsed(), 1);
+    /// ```
+    #[inline]
+    pub fn elapsed(&self) -> u64 {
+        self.elapsed
+    }
+
+    /// Sets the elapsed time of the stopwatch.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut stopwatch = DiscreteStopwatch::new();
+    /// stopwatch.set_elapsed(1);
+    /// assert_eq!(stopwatch.elapsed(), 1);
+    /// ```
+    #[inline]
+    pub fn set_elapsed(&mut self, time: u64) {
+        self.elapsed = time;
+    }
+
+    /// Advance the stopwatch by `delta` increments.
+    /// If the stopwatch is paused, ticking will not have any effect
+    /// on elapsed time.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut stopwatch = DiscreteStopwatch::new();
+    /// stopwatch.tick(3);
+    /// assert_eq!(stopwatch.elapsed(), 3);
+    /// ```
+    pub fn tick(&mut self, delta: u64) -> &Self {
+        if !self.paused() {
+            self.elapsed += delta;
+        }
+        self
+    }
+
+    /// Pauses the stopwatch. Any call to [`tick`](Stopwatch::tick) while
+    /// paused will not have any effect on the elapsed time.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut stopwatch = DiscreteStopwatch::new();
+    /// stopwatch.pause();
+    /// stopwatch.tick(2);
+    /// assert!(stopwatch.paused());
+    /// assert_eq!(stopwatch.elapsed(), 0);
+    /// ```
+    #[inline]
+    pub fn pause(&mut self) {
+        self.paused = true;
+    }
+
+    /// Unpauses the stopwatch. Resume the effect of ticking on elapsed time.
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut stopwatch = DiscreteStopwatch::new();
+    /// stopwatch.pause();
+    /// stopwatch.tick(1);
+    /// stopwatch.unpause();
+    /// stopwatch.tick(1);
+    /// assert!(!stopwatch.paused());
+    /// assert_eq!(stopwatch.elapsed(), 1);
+    /// ```
+    #[inline]
+    pub fn unpause(&mut self) {
```

Comment:
What about `continue` in favor of `unpause`?

Reply:
I'd probably call that out-of-scope for this PR; I'm just following the existing API :)

(I also slightly prefer `unpause`, and think `resume` is also better than `continue` if we want to change.)